application/classes/Ushahidi/Repository/TargetedSurveyState.php

```diff
@@ -47,9 +47,15 @@ public function getByPost($post)
 		return new Entity\Post($this->selectOne(compact('post')));
 	}
 
-	public function getByContactId($contact_id)
+	public function getActiveByContactId($contact_id)
 	{
-		return $this->getEntity($this->selectOne(compact('contact_id')));
+		return $this->getEntity($this->selectOne([
+			'contact_id' => $contact_id,
+			'survey_status' => [
```

Comment:
ohhhh! awesome.

Reply:
@rjmackay FYI : needs to be integrated and ported to lumen .